VideoLocker/src/main/java/org/edx/mobile/module/db/impl/DbHelper.java

```diff
@@ -109,12 +116,60 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
                 createAssessmentTable(db);
             }
 
+            if (oldVersion < 6) {
+                Cursor cursor = db.query(false, DbStructure.Table.DOWNLOADS,
+                        new String[]{DbStructure.Column.ID, DbStructure.Column.USERNAME,
+                                DbStructure.Column.FILEPATH}, null, null, null, null, null, null);
+                if (cursor != null) {
+                    final File appExternalDir = mContext.getExternalFilesDir(null).getParentFile();
+                    try {
+                        while (cursor.moveToNext()) {
+                            final int id = cursor.getInt(0);
+                            final String username = cursor.getString(1);
+                            final String filePath = cursor.getString(2);
+                            final String encryptedUsername = Sha1Util.SHA1(username);
+                            final String newFilePath = filePath.replace(
+                                    appExternalDir.getAbsolutePath() + "/" + username,
+                                    appExternalDir.getAbsolutePath() + "/"
+                                            + AppConstants.Folders.VIDEOS + "/"
+                                            + encryptedUsername);
+
+                            // First update the folder names
+                            final File previousFolder = new File(appExternalDir, username);
+                            if (previousFolder.exists()) {
+                                final File newFolder = new File(appExternalDir,
+                                        AppConstants.Folders.VIDEOS + "/" + encryptedUsername);
+                                if (!newFolder.exists()) newFolder.mkdirs();
+                                previousFolder.renameTo(newFolder);
+                            }
+
+                            // Then update the database rows
+                            final ContentValues updatedValues = new ContentValues();
+                            updatedValues.put(DbStructure.Column.USERNAME, encryptedUsername);
+                            updatedValues.put(DbStructure.Column.FILEPATH, newFilePath);
+                            db.update(DbStructure.Table.DOWNLOADS, updatedValues,
+                                    DbStructure.Column.ID + "=" + id, null);
```

Comment:
You should bind the `id` parameter by passing it as a string array (with a single item) in the last parameter here, so that the statement doesn't change each time, and can thus be compiled to a prepared statement by the SQLite session implementation. Note that you can get it directly as a string from the `Cursor` itself if you want.